marketplace-builder/src/builder.rs

```diff
@@ -502,8 +506,47 @@ mod test {
             .unwrap()
     }
 
-    #[async_std::test]
-    async fn test_marketplace_reserve_builder() {
+    /// Verify that the bundle associated with the received quorum proposal contains the expected
+    /// transaction.
+    ///
+    /// # Returns
+    /// * `true` if the verification succeeds.
+    /// * `false` if the expected quorum proposal isn't received yet.
+    ///
+    /// # Panics
+    /// * If the verification fails.
+    async fn verify_bundle(
```

Comment:
Nit: I think the control flow would be a bit clearer if this function returned Option<Vec<Transactions>> and assert would be relegated to the caller

Reply:
Good suggestion. I had to change it to `get_bundle` rather than `get_transactions` as you suggested, in order to make it compatible with the fee assertions recently added to the tests.

However, as mentioned in [this comment](https://github.com/EspressoSystems/espresso-sequencer/pull/2011/files#r1776047858), I also changed the logic a bit to handle empty transactions the same as non-empty transactions as long as the bundle fetching succeeds. This caused the public mempool tests to fail the transaction assertion.